Foundation/Operation.swift

```diff
@@ -234,81 +234,163 @@ extension OperationQueue {
     public static let defaultMaxConcurrentOperationCount: Int = Int.max
 }
 
+internal class _IndexedOperationLinkedList {
+    
+    internal class Node {
+        var operation: Operation
+        var next: _IndexedOperationLinkedList.Node? = nil
+        weak var previous: _IndexedOperationLinkedList.Node? = nil
+        
+        init(operation: Operation) {
+            self.operation = operation
+        }
+    }
+    
+    private(set) var root: _IndexedOperationLinkedList.Node? = nil
+    private(set) weak var tail: _IndexedOperationLinkedList.Node? = nil
+    private(set) var count: Int = 0
```

Comment:
is this really needed? can't we just keep track of the count externally in the OperationQueue?

Reply:
I see your point here. I think I would be useful to have the count of operations for each priorities, may be in the future. 

If you think it don't haven any need now, as well as in the future, then it totally makes sense to remove this property from the linked list and move it over to `OperationQueue`.

Reply:
I thought about this more, this property don't have any use inside `_IndexedOperationLinkedList`. I would make much more sense if the property `count` is moved inside `_OperationList` structure, as we just need to keep counting the operations just in `_OperationList.all`.

Reply:
Resolving this comment since it is addressed